src/bbcore.js

```diff
@@ -71,10 +71,30 @@ function BBCore(options) {
     this.__vidRecording = false;
     this.__vidRecHndl = null;
 
+    // polyfill for Object.create.
+    if (typeof Object.create !== "function") {
+        Object.create = function (proto, propertiesObject) {
+            if (typeof proto !== 'object' && typeof proto !== 'function') {
+                throw new TypeError('Object prototype may only be an Object: ' + proto);
+            } else if (proto === null) {
+                throw new Error("This browser's implementation of Object.create is a shim and doesn't support 'null' as the first argument.");
+            }
+
+            if (typeof propertiesObject != 'undefined') {
+                throw new Error("This browser's implementation of Object.create is a shim and doesn't support a second argument.");
+            }
+
+            function F() {}
+            F.prototype = proto;
+
+            return new F();
+        };
+    }
+
     /** @class */
     this.contacts = function () {
     };
-    this.contacts.prototype = Array.prototype;
+    this.contacts.prototype = Object.create(Array.prototype);
```

Comment:
We have also been affected by this, since the `.find` implementation here is not consistent with native find. As it stands, we're blocking this implementation by overriding property descriptors, but a solution such as this would be much more ideal.

That said, Instead of providing a partial polyfill for Object.create, it _might_ be worth just targeting the consumption directly.

```javascript
this.contacts.prototype = [];
this.videos.prototype = [];
```